spec.html

```diff
@@ -8528,7 +8524,7 @@ <h1>GeneratorFunctionCreate ( _kind_, _ParameterList_, _Body_, _Scope_ )</h1>
       <p>The abstract operation GeneratorFunctionCreate requires the arguments: _kind_ which is one of (~Normal~, ~Method~), a parameter list Parse Node specified by _ParameterList_, a body Parse Node specified by _Body_, a Lexical Environment specified by _Scope_. GeneratorFunctionCreate performs the following steps:</p>
       <emu-alg>
         1. Let _functionPrototype_ be %Generator%.
-        1. Let _F_ be FunctionAllocate(_functionPrototype_, ~generator~).
+        1. Let _F_ be FunctionAllocate(_functionPrototype_, *false*).
```

Comment:
I would much prefer a descriptive enum over booleans. How about `~NeedsConstruct~` and `~NoConstruct~`?

Reply:
> I would much prefer a descriptive enum over booleans.

(Personally, I think I'd prefer for ECMASpeak to have named parameters, e.g.
```Let _F_ be FunctionAllocate(..., _needsConstruct_=*true*).```
 but I guess a boolean-like enum is more likely to happen.)

> How about `~NeedsConstruct~` and `~NoConstruct~`?

Possibly. Though it's more common for spec-values to start with a lower-case letter.

Other possibilities:
- `~addConstructMethod~` and `~dontAddConstructMethod~`
- `~makeItAConstructor~` and `~dontMakeItAConstructor~`
- `~isAConstructor~` and `~isntAConstructor~`
- `~constructor~` and `~notConstructor~`

Reply:
Re: named parameters, that's a more pervasive change so let's punt on it for now.

Out of other possible enum options, `~addConstructMethod~` and `~dontAddConstructMethod~` is also fine, but I remain partial to `~needsConstruct~` and `~noConstruct~` for brevity.

Mind changing it to either of those options? I'm fine with either.

Reply:
I'll take the editors' direction on this.

Reply:
(Technically not a slots list, but rather a list of internal method names. But by analogy with a slots list, the name should include the double square brackets.)

It wouldn't be enough to assert that the list contains (at most) those two names, because that wouldn't prevent invoking the operation with `<< Construct >>`, which wouldn't make any sense. So we'd have to assert that the value is either '<< Call, Construct >>` or `<< Call >>`. (And then not care about the presence of `Call`.)

---

I'm starting to think that maybe we shouldn't be passing an arg for this to `FunctionAllocate` in the first place. We could just have `FunctionAllocate` create a non-constructor function, and then, if a constructor is wanted, make it so after the fact. (Note that there's only two spots where a constructor might be wanted.)

Reply:
That seems reasonable to me as well.

Reply:
In fact, I've now convinced myself that:
- the set of functions for which `FunctionAllocate` adds `[[Construct]]`. and
- the set of functions for which `MakeConstructor` is called

are exactly the same. It seems very odd to me that these two constructor-making things happen separately.